eth/stagedsync/stage_interhashes.go

```diff
@@ -98,11 +97,18 @@ func SpawnIntermediateHashesStage(s *StageState, u Unwinder, tx kv.RwTx, cfg Tri
 		if cfg.checkRoot && root != expectedRootHash {
 			log.Error(fmt.Sprintf("[%s] Wrong trie root of block %d: %x, expected (from header): %x. Block hash: %x", logPrefix, to, root, expectedRootHash, headerHash))
 			if cfg.badBlockHalt {
-				return trie.EmptyRoot, fmt.Errorf("Wrong trie root")
+				return trie.EmptyRoot, fmt.Errorf("wrong trie root")
 			}
 			if cfg.hd != nil {
-				header := rawdb.ReadHeader(tx, headerHash, to)
-				cfg.hd.ReportBadHeaderPoS(headerHash, header.ParentHash)
```

Comment:
@enriavil1 - above I see code:
```
headerHash = syncHeadHeader.Hash()
```
- so, why do we again read this header from db? 
- if we just did read this header from db by method .HeaderByNumber(), why can't we read it again by .HeaderByHash()?

Reply:
Actually we read that header from `blockReader` as well. What I didn't notice is that we actually have `syncHeadHeader` meaning that there is no reason for us to be reading the header once again.

Reply:
probably we need one more pr about it

Reply:
yes I just created a branch deleting the re-fetching of the header from snapshots/db should I open it?

Reply:
create new one - this one merged already

Reply:
done